src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,106 @@
+import { HttpClient, HttpEvent, HttpEventType } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { apiSettings } from '../../environments/apisettings';
+import { MoreResults, RecordPublicationInfo, Results, SortParameter } from '../_models';
+
+import { collectResultsUptoPage } from './service-utils';
+import { ErrorService } from './error.service';
+
+@Injectable({ providedIn: 'root' })
+export class DepublicationService {
+  constructor(private readonly http: HttpClient, private readonly errors: ErrorService) {}
+
+  /** handleUploadEvents
+  /*  record upload events
+  /*  @param {HttpEvent} event - event from the server
+  */
+  handleUploadEvents(event: HttpEvent<Object>): boolean {
+    if (event.type === HttpEventType.Response) {
+      return true;
+    } else {
+      if (event.type === HttpEventType.UploadProgress) {
+        console.log(`File upload: ${event.loaded} / ${event.total}`);
+      }
+      return false;
+    }
+  }
+
+  /** setPublicationFile
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {File} file - file of record urls
+  */
+  setPublicationFile(datasetId: string, file: File): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/records/${datasetId}`;
```

Comment:
Can we rename `.../records/...` to `.../depublished_records/...`? Perhaps one day we wish to add other records for some reason.